tests/ftests/067-sudo-systemd_cgexec-v1.py

```diff
@@ -0,0 +1,199 @@
+
+
+def setup(config):
+    result = consts.TEST_PASSED
+    cause = None
+
+    Systemd.write_config_with_pid(config, CONFIG_FILE_NAME, SLICE, SCOPE)
+
+    Cgroup.configparser(config, load_file=CONFIG_FILE_NAME)
+
+    # create and check if the cgroup was created under the systemd default path
+    if not Systemd.create_and_validate(config, CONTROLLER, SYSTEMD_CGNAME):
+        result = consts.TEST_FAILED
+        cause = (
+                    'Failed to create systemd delegated cgroup {} under '
+                    '/sys/fs/cgroup/{}/{}/{}/'.format(SYSTEMD_CGNAME, CONTROLLER, SLICE, SCOPE)
+                )
+        return result, cause
+
+    # create and check if the cgroup was created under the controller sub-tree
+    if not Systemd.create_and_validate(config, CONTROLLER, OTHER_CGNAME, ignore_systemd=True):
+        result = consts.TEST_FAILED
+        cause = (
+                    'Failed to create cgroup {} under '
+                    '/sys/fs/cgroup/{}/'.format(OTHER_CGNAME, CONTROLLER)
+                )
+
+    return result, cause
+
+
+def create_process_get_pid(config, CGNAME, SLICE='', ignore_systemd=False):
+    result = consts.TEST_PASSED
+    cause = None
+
+    config.process.create_process_in_cgroup(
+                                                config, CONTROLLER, CGNAME, cgclassify=False,
+                                                ignore_systemd=ignore_systemd
+                                            )
+
+    # We need pause, before the cgroups.procs gets updated, post cgexec
+    time.sleep(1)
+
+    pids = Cgroup.get_pids_in_cgroup(config, os.path.join(SLICE, CGNAME), CONTROLLER)
+    if pids is None:
+        result = consts.TEST_FAILED
+        cause = 'No processes were found in cgroup {}'.format(CGNAME)
+
+    return pids, result, cause
+
+
+def terminate_process(config, pids):
+    if pids:
+        for p in pids.splitlines():
+            Run.run(['sudo', 'kill', '-9', p])
+
+
+def test(config):
+    global SYSTEMD_PIDS, OTHER_PIDS
+
+    result = consts.TEST_PASSED
+    cause = None
+
+    # Test cgclassify, that creates a process and then uses cgclassify
+    # to migrate the task the cgroup.
+    SYSTEMD_PIDS, result, cause = create_process_get_pid(
+                                                            config, SYSTEMD_CGNAME,
+                                                            os.path.join(SLICE, SCOPE)
+                                                        )
+    if result == consts.TEST_FAILED:
+        return result, cause
+
+    OTHER_PIDS, result, tmp_cause = create_process_get_pid(
+                                                                config, OTHER_CGNAME,
+                                                                ignore_systemd=True
+                                                          )
+    if result == consts.TEST_FAILED:
+        return result, cause
+
+    # classify the task from the non-systemd scope cgroup to systemd scope
```

Comment:
I don't understand the need to classify the processes to another cgroup.  Isn't the point of this test to run cgexec?  If so, wouldn't a clearer validation simply be to verify that the process(es) were created in the requested cgroups?

Reply:
Agreed, just checking for process creation under the right cgroup should be sufficient and the migration test is not required here